fcrepo-kernel/src/main/java/org/fcrepo/kernel/utils/NamespaceTools.java

```diff
@@ -90,6 +94,12 @@ public static void validatePath(final Session session, final String path) {
         final String relPath = path.replaceAll("^/+", "").replaceAll("/+$", "");
         final String[] pathSegments = relPath.split("/");
         for (final String segment : pathSegments) {
+            // Empty path elements are not valid!
```

Comment:
No, this is the wrong place for this test. Putting it here assumes that HTTP URIs are translated wholemeal into JCR paths, which is the exact assumption that the `IdentifierConverter` system was designed to break. This test belongs in `HttpResourceConverter` or near it.